### PR TITLE
508 search filter

### PR DESCRIPTION
**Pull request by flesher1**
Adds 508 Search flag
Improve routing within search
508 flag sharable from share search
Adds default profile search flag
Adds link to listing links for consistent sharability.

test filter, default option from profile and listing page.
Note, you will need to add 508 to several listing to test.

closes AMLOS=609

### CODE REVIEW

File: src/components/navBar/SearchFilter.js [hunk 1]
```diff
@@ -14,9 +15,25 @@ class SearchFilter extends React.Component {
...
+            this.setState({accessableOnlySearch: true})
+        }else if(this.props.filter.accessableOnlySearch !== null && this.props.filter.accessableOnlySearch !== "None"){
+            //looking for true and "true"
+            this.setState({accessableOnlySearch: true})
+        }
     }
 
     shouldComponentUpdate(nextProps, nextState){
+        if(this.props.profile.accessableOnlySearch === nextProps.profile.accessableOnlySearch){
+            return true;
```
Comment (irtiza-rizvi) [inline src/components/navBar/SearchFilter.js]: Should the check be for `this.props.profile.accessableOnlySearch !== nextProps.profile.accessableOnlySearch` instead?

**Comment by irtiza-rizvi** [comment]
You're going to hate me for this, but the correct spelling is `accessible`.

File: src/components/navBar/SearchFilter.js (hunk 1 above)
Comment (flesher1) [reply src/components/navBar/SearchFilter.js]: So I put that in there for a particular issue, but I cannot remember what the issue was. Im going to   make the change and hope it works as intended.

**Comment by irtiza-rizvi** [comment]
Changes look good :+1:

**Comment by blynch11p** [comment]
:+1:

**Pull request merged**